### PR TITLE
amd support

### PR DESCRIPTION
**Pull request by heldr**
Hi Leo! What's up?

I've implemented an _amd_ support following the same way jQuery does: Running not only through `window.Dexter` but also with `require(['dexter'])` call.

I also ran npm update throughout devDependencies and created a dist source map file.

Cheers!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,23 +31,29 @@
...
+    "grunt-contrib-jshint": "~0.8.0",
+    "grunt-contrib-nodeunit": "~0.3.2",
+    "grunt-contrib-uglify": "~0.3.2",
+    "grunt-contrib-watch": "~0.5.3",
+    "grunt-coveralls": "~0.3.0",
+    "grunt-jscs-checker": "~0.4.0",
+    "grunt-qunit-istanbul": "~0.3.0",
+    "mdoc": "~0.3.4",
+    "qunitjs": "~1.14.0",
+    "requirejs": "~2.1.11"
```
Comment (leobalter) [inline package.json]: would you remove the tildes (`~`)?

Regarding a sturdy build process I prefer to make strict references here.

**Comment by leobalter** [comment]
I do not want to support many amd/commonjs kinds before it's not consolidated or specified on ES6.

Even though, requirejs looks very popular and there's still a long way to the ES6 modules.

I'm accepting this PR after you making the changes. Would you also rename your commit msg to `Adds requirejs support`? 

Thank you!

**Comment by leobalter** [comment]
Thank you!

**Comment by heldr** [comment]
np!

**Pull request merged**